src/App.tsx

```diff
@@ -1,203 +1,75 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Header } from './components/Header';
+import { Main } from './components/Main';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { Errors } from './components/Errors';
+import { Status } from './types/enums';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterSelected, setFilterSelected] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [errorHidden, setErrorHidden] = useState(true);
+
+  function showError() {
+    setErrorHidden(false);
+
+    setTimeout(() => {
+      setErrorHidden(true);
+    }, 2000);
+  }
+
+  useEffect(() => {
+    getTodos()
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+      })
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        showError();
+      });
+  }, []);
```

Comment:
1. **Dependencies Usage:**
   Recommend using `useCallback` for `showError` to ensure stable references and adding it to the dependency array.

   ```javascript
   const showErrorCallback = useCallback(() => {
     // showError logic
   }, []); // Add dependencies if needed

   // ...

   useEffect(() => {
     // Some effect using showErrorCallback
   }, [showErrorCallback]);
   ```

2. **Error Handling Logic:**
   Suggest clearing the timeout on component unmount to avoid potential memory leaks.

   ```javascript
   useEffect(() => {
     const timeoutId = setTimeout(() => {
       // Some logic
     }, 3000);

     return () => clearTimeout(timeoutId); // Clear timeout on unmount
   }, [/* dependencies */]);
   ```